rapids_logger/CMakeLists.txt

```diff
@@ -0,0 +1,215 @@
+  configure_file(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/logger.hpp.in ${LOGGER_OUTPUT_FILE})
+  install(FILES ${LOGGER_OUTPUT_FILE} DESTINATION ${INSTALL_DIR})
+
+  set(LOGGER_IMPL_OUTPUT_FILE ${BUILD_DIR}/logger_impl/logger_impl.hpp)
+  configure_file(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/logger_impl.hpp.in ${LOGGER_IMPL_OUTPUT_FILE})
+  install(FILES ${LOGGER_IMPL_OUTPUT_FILE} DESTINATION ${INSTALL_DIR}/logger_impl)
+
+  add_library(${_RAPIDS_LOGGER_TARGET} INTERFACE)
+  include(GNUInstallDirs)
+  # Note: The BUILD_INTERFACE setting assumes that LOGGER_HEADER_DIR is the subdirectory of
+  # CMAKE_INSTALL_INCLUDEDIR relative to which all includes are rooted in the C++ code files. I
+  # think that is a safe assumption though since if it were violated then the INSTALL_INTERFACE
+  # would not only be incorrect (if computed using LOGGER_HEADER_DIR), but it would also break
+  # consumers of the installed package who expect to be able to write `#include
+  # <${LOGGER_HEADER_DIR/include\//}/logger.hpp>` and have it work.
+  target_include_directories(
+    ${_RAPIDS_LOGGER_TARGET}
+    INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_INSTALL_INCLUDEDIR}>"
+              "$<INSTALL_INTERFACE:${_RAPIDS_LOGGER_HEADER_DIR}>")
+  if(_RAPIDS_RMM_BACKWARDS_COMPATIBILITY)
+    if(NOT _RAPIDS_SUPPORTS_LOGGING)
+      message(STATUS "RMM_BACKWARDS_COMPATIBILITY requires SUPPORTS_LOGGING, turning it on")
+      set(_RAPIDS_SUPPORTS_LOGGING ON)
+    endif()
+    target_compile_definitions(${_RAPIDS_LOGGER_TARGET} INTERFACE RMM_BACKWARDS_COMPATIBILITY)
+  endif()
+  if(_RAPIDS_CUDF_BACKWARDS_COMPATIBILITY)
+    target_compile_definitions(${_RAPIDS_LOGGER_TARGET} INTERFACE CUDF_BACKWARDS_COMPATIBILITY)
+  endif()
+  if(_RAPIDS_SUPPORTS_LOGGING)
+    target_compile_definitions(${_RAPIDS_LOGGER_TARGET} INTERFACE SUPPORTS_LOGGING)
+
+    if(NOT _RAPIDS_RMM_BACKWARDS_COMPATIBILITY)
+      # Create an interface target that will trigger compilation of the logger implementation in any
+      # target that is linked to it.
+      set(LOGGER_IMPL_SRC_OUTPUT_FILE ${BUILD_DIR}/logger_impl/logger.cpp)
+      configure_file(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/logger.cpp.in
+                     ${LOGGER_IMPL_SRC_OUTPUT_FILE})
+      install(FILES ${LOGGER_IMPL_SRC_OUTPUT_FILE} DESTINATION ${INSTALL_DIR}/logger_impl)
+
+      # Note that we cannot specify the source files directly in add_library, see the CMake
+      # documentation explaining that these do not populate INTERFACE_SOURCES.
+      # https://cmake.org/cmake/help/latest/command/add_library.html#interface-with-sources
+      set(impl_target ${_RAPIDS_LOGGER_TARGET}_impl)
+      add_library(${impl_target} INTERFACE)
+      target_sources(
+        ${impl_target}
+        INTERFACE $<BUILD_INTERFACE:${LOGGER_IMPL_SRC_OUTPUT_FILE}>
+                  $<INSTALL_INTERFACE:${_RAPIDS_LOGGER_HEADER_DIR}/logger_impl/logger.cpp>)
+      target_link_libraries(${impl_target} INTERFACE ${_RAPIDS_LOGGER_TARGET})
```

Comment:
This target requires `INTERFACE` on the spdlog target since the `logger.cpp` uses headers from spdlog

Reply:
Whoops I lost that at some point, thanks.

Reply:
Ah I see what happened. There's a bit of additional complexity on whether the impl or the public target should get the linkage depending on whether we're in compatibility mode or not.